rubicon/objc/eventloop.py

```diff
@@ -569,6 +569,32 @@ def _new_default_loop(self):
         loop._policy = self
         return loop
 
+    def _init_watcher(self):
+        with events._lock:
+            if self._watcher is None:  # pragma: no branch
+                self._watcher = SafeChildWatcher()
+                if isinstance(threading.current_thread(), threading._MainThread):
```

Comment:
Fair call. This implementation was cribbed from Python's own internal implementation - but AFAICT there's no reason not to just check if "current thread is main thread", rather than the type check.